### ISSUE TITLE
KeyError from Flask when the DD SDK is init

### ISSUE THREAD

**Issue by JordanP** [DataDog/dd-trace-py#2549]
### Which version of dd-trace-py are you using?
I am upgrading from 0.29.0 to 0.49.3 (I know....)

### Which version of pip are you using?
A recent one.

### Which version of the libraries are you using?

You can copy/paste the output of `pip freeze` here.
- click==7.1.2
- flask==0.12.4 
- werkzeug==0.14.1
### How can we reproduce your problem?
I am running Nginx in front of uwsgi serving a Flask app.
My Nginx conf has:
```
add_header X-Request-ID $request_id always;
include uwsgi_params;
uwsgi_param HTTP_X-Request-ID $request_id;
```

I am *not* wrapping uwsgi with ddtrace-run 
### What is the result that you get?
```
spawned uWSGI master process (pid: 1)
spawned uWSGI worker 1 (pid: 6, cores: 2)
*** Stats server enabled on :9099 fd: 9 ***
running "unix_signal:15 gracefully_kill_them_all" (master-start)...
WSGI app 0 (mountpoint='') ready in 8 seconds on interpreter 0x55e16f4bc7a0 pid: 6 (default app)
running "echo 1 > /tmp/uwsgi-ready" (post app)...
Traceback (most recent call last):
  File "/gorgias/.local/lib/python3.7/site-packages/sentry_sdk/integrations/flask.py", line 87, in sentry_patched_wsgi_app
    environ, start_response
  File "/gorgias/.local/lib/python3.7/site-packages/sentry_sdk/integrations/wsgi.py", line 140, in __call__
    reraise(*_capture_exception(hub))
  File "/gorgias/.local/lib/python3.7/site-packages/sentry_sdk/_compat.py", line 54, in reraise
    raise value
  File "/gorgias/.local/lib/python3.7/site-packages/sentry_sdk/integrations/wsgi.py", line 136, in __call__
    _sentry_start_response, start_response, transaction
  File "/gorgias/.local/lib/python3.7/site-packages/sentry_sdk/integrations/flask.py", line 86, in <lambda>
    return SentryWsgiMiddleware(lambda *a, **kw: old_app(self, *a, **kw))(
  File "/gorgias/.local/lib/python3.7/site-packages/flask/app.py", line 1997, in __call__
    return self.wsgi_app(environ, start_response)
  File "/gorgias/.local/lib/python3.7/site-packages/ddtrace/contrib/flask/helpers.py", line 25, in wrapper
    return func(pin, wrapped, instance, args, kwargs)
  File "/gorgias/.local/lib/python3.7/site-packages/ddtrace/contrib/flask/patch.py", line 329, in traced_wsgi_app
    request_headers=request.headers,
  File "/gorgias/.local/lib/python3.7/site-packages/ddtrace/contrib/trace_utils.py", line 254, in set_http_meta
    _store_request_headers(dict(request_headers), span, integration_config)
  File "/gorgias/.local/lib/python3.7/site-packages/werkzeug/datastructures.py", line 1354, in __getitem__
    return _unicodify_header_value(self.environ['HTTP_' + key])
KeyError: 'HTTP_X_REQUEST_ID'
```
### What is the result that you expected?
A working app :)

**Comment by brettlangdon** [comment]
Hey @JordanP I am sorry you are having issues. Are you able to share the traceback/`KeyError` you are getting?

**Comment by brettlangdon** [comment]
> I hit the `post` button too fast... I have edited the issue.

No worries, it did look a little off. Thanks for sharing.

**Comment by brettlangdon** [comment]
It looks like werkzeug doesn't like `uwsgi_param HTTP_X-Request-ID $request_id;`, since when you try to access a value off of the request headers (`EnvironHeaders`) it'll transform the key provided into a WSGI standard format header for environ and try to access.

https://github.com/pallets/werkzeug/blob/6fc07fec6dfd21690c9909f8c7c8e744d55afba7/werkzeug/datastructures.py#L1351

Now, this is definitely initiated by `ddtrace` since we convert the request headers into a `dict` for processing, so we are the ones trying to access a non-existent header in the environ.

I have opened #2550 since I noticed a small optimization we could make (only call this code if we need to), but that still doesn't solve the case if you wanted to use our header tracing feature (add request/response headers as tags on the span).

I'll keep looking if there is a good solution for us here. You could always change `uwsgi_param HTTP_X-Request-ID` -> `uwsgi_param HTTP_X_Request_ID` to match WSGI standards, but we should still be able to handle this case/type of issue anyways.

**Comment by JordanP** [comment]
So #2550 would work for me, because we are not interested in header tracing feature... yet  

But I need a new release to be cut :p

**Comment by brettlangdon** [comment]
Thanks @JordanP, I am still going to think about how we can make this piece of code more resilient, just in case someone else runs into the same case.

Once #2550 is merged I can get you an RC release to use. (Maybe later today or tomorrow at latest)

**Comment by brettlangdon** [comment]
@JordanP we have just released a prerelease that has this change: `pip install ddtrace==0.50.0rc2`.

**Comment by brettlangdon** [comment]
This has now been released in `ddtrace==0.50.0`
